core/src/main/scala/org/apache/spark/sql/sources/StoreStrategy.scala

```diff
@@ -52,9 +52,9 @@ object StoreStrategy extends Strategy {
     case DMLExternalTable(name, storeRelation: LogicalRelation, insertCommand) =>
       ExecutedCommand(ExternalTableDMLCmd(storeRelation, insertCommand)) :: Nil
 
-    case InsertIntoTable(l @ LogicalRelation(t: RowPutRelation, _),
-    part, query, overwrite, false) if part.isEmpty =>
-      ExecutedCommand(PutIntoDataSource(l, query, overwrite)) :: Nil
+    case InsertIntoTable(l@LogicalRelation(t: RowPutRelation, _),
```

Comment:
This kind of fixes the issue seen in SNAP-677 because now PutIntoDataSource will only be invoked if it is a insertInto with SaveMode==Overwrite which will not happen. However, as we discussed we don't want to change the behaviour of overwrite to be like put (rather than a truncate which is what happens for other data sources). This is handled by the other PR I have created.